### PR TITLE
[MRG] New widget to adjust default smoothing value in gui

### PR DESCRIPTION
**Pull request by dylansdaniels**
Title says it all

### CODE REVIEW

File: hnn_core/gui/_viz_manager.py [hunk 1]
```diff
@@ -902,6 +906,9 @@ def __init__(self, gui_data, viz_layout):
             button_style="primary",
             style={'button_color': self.viz_layout['theme_color']},
             layout=self.viz_layout['btn'])
+        # self.make_fig_button.on_click(
+        #     lambda b: self.add_figure(self.default_smoothing)
+        # )
```
Comment (ntolley) [inline hnn_core/gui/_viz_manager.py:911]:
```suggestion
```

File: hnn_core/gui/_viz_manager.py [hunk 2]
```diff
@@ -937,6 +944,11 @@ def data(self):
             "figs": self.figs
         }
 
+    @property
+    def fig_default_params(self):
```
Comment (ntolley) [inline hnn_core/gui/_viz_manager.py:948]: Does this need to be `@property`? Seems like the other ones were created to regroup existing attributes for code cleanliness, whereas here you're just returning `self.fig_defaults`

File: hnn_core/tests/test_gui.py [hunk 3]
```diff
@@ -1116,3 +1116,53 @@ def test_delete_single_drive(setup_gui):
...
+    new_smoothing = 0
+    gui.widget_default_smoothing.value = new_smoothing
+
+    gui.run_button.click()
+
+    # check that new default smoothing value is set everywhere
+    gui_smooth_value = gui.fig_default_params['default_smoothing']
+    viz_smooth_value = gui.viz_manager.fig_default_params['default_smoothing']
+
+    assert gui_smooth_value == new_smoothing
```
Comment (ntolley) [inline hnn_core/tests/test_gui.py:1139]: It'd be useful to add a similar assert for the original smooth values of the `fig_default` and `viz_manager`

File: hnn_core/gui/_viz_manager.py (hunk 2 above)
Comment (dylansdaniels) [reply hnn_core/gui/_viz_manager.py:948]: I made it a @property so I could call it in the test:
```viz_smooth_value = gui.viz_manager.fig_default_params['default_smoothing']```

File: hnn_core/gui/_viz_manager.py (hunk 1 above)
Comment (dylansdaniels) [reply hnn_core/gui/_viz_manager.py:911]: Good catch 😅

**Review by asoplata — APPROVED** [review]
LGTM. Tested and works, even for subsequent runs in-session

**Pull request merged**